src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -251,11 +249,12 @@ internal void StartMessageBatchReceiver(CancellationToken cancellationToken)
                             return;
                         }
 
-                        if (_isSessionsEnabled)
+                        if (_isSessionsEnabled && ( receiver == null || receiver.IsClosedOrClosing))
                         {
                             try
                             {
                                 receiver = await sessionClient.AcceptMessageSessionAsync();
+                                receiver.PrefetchCount = _serviceBusOptions.PrefetchCount;

```

Comment:
There are some rare cases where this returns null. It was a bug long time ago, not sure if its fixed or not. For the sake of precautions, I would put a null check after this and if this happens to be null, do a `continue`

Reply:
There will be a NullReferenceException and we will try to accept a new session on the next loop iteration.